server/src/main/java/org/opensearch/repositories/blobstore/BlobStoreRepository.java

```diff
@@ -851,6 +857,7 @@ private void doDeleteShardSnapshots(
                 ActionListener.wrap(() -> listener.onResponse(updatedRepoData)),
                 2
             );
+            releaseResourceLockFiles(snapshotIds, repositoryData, remoteStoreLockManagerFactory, afterCleanupsListener);
```

Comment:
1. This will break the existing mechanism as `afterCleanupsListener` has a group size of 2 whereas 3 different actions are now getting invoked.

2. It seems like this can be extracted in a separate delete API instead of current one to prevent the API compatibility issue. After the `RepositoryData` is updated, the blobs are cleaned up and then invoke the listener with updated `RepositoryData`. The other API can be called with this updated `RepositoryData` and perform the cleanup of the remote store shard level files which will invoke original `listener` on success.

Reply:
Added these changes in the revision